modules/updater/test/updater.test.ts

```diff
@@ -50,6 +50,14 @@ describe("update() can immutably update objects using", () => {
       });
     });
 
+    it("returns the same object when the result is unchanged", () => {
+      const obj = { name: { first: "Shin", last: "Doe" } };
+      const operation = { $set: { "name.first": "Shin" } };
+
+      const updatedObject = update(obj, operation);
+      assert.strictEqual(obj, updatedObject);
```

Comment:
`strictEqual` doesn't mean same as `===`.

`obj` has same property to `updatedObject` but is different instance from `updatedObject`.
This cause strictEqual return `true` but `obj === updatedObject` return `false`.

The difference cause re-render for [react-redux](https://react-redux.js.org/api/connect#arestatesequal-next-object-prev-object-boolean)

The test doesn't ensure the issue fixed.

```suggestion
      assert.strictEqual(obj === updatedObject, true);
```

Reply:
Spec of `assert.strictEqual` is here.
https://nodejs.org/api/assert.html#assert_assert_strictequal_actual_expected_message

This says 
> Tests strict equality between the actual and expected parameters as determined by the SameValue Comparison.

SameValue Comparison is determined by ECMA262.
https://tc39.es/ecma262/#sec-samevalue

For object equality, it uses `! SameValueNonNumeric(x, y).`
The algorithm is here.
https://tc39.es/ecma262/#sec-samevaluenonnumeric
It says
> If x and y are the same Object value, return true. Otherwise, return false.

If you doubt the spec, please check the address equalities at these test cases, which were already checked by myself.

Reply:
I mistook `strictEqual` to `deepEqual`. Sorry for that and I confirm tests fixed the issue.